Source/Private/PerfGrapherCommandlet.cpp

```diff
@@ -90,23 +108,94 @@ int32 UPerfGrapherCommandlet::Main( const FString & params )
 
     for ( const auto & package_name : package_names )
     {
-        auto * package = LoadPackage( nullptr, *package_name, 0 );
-        if ( package == nullptr )
+        UE_LOG( LogPerfGrapher, Log, TEXT( "Processing package: %s" ), *package_name );
+        if ( !RunPerfGrapher( package_name, metrics_params ) )
         {
-            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not load package %s" ), *package_name );
-            return 2;
+            UE_LOG( LogPerfGrapher, Error, TEXT( "Failed to process map %s" ), *package_name );
+            return 1;
         }
+    }
 
-        auto * world = UWorld::FindWorldInPackage( package );
-        if ( world == nullptr )
-        {
-            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not get a world in the package %s" ), *package_name );
-            return 2;
-        }
+    UE_LOG( LogPerfGrapher, Log, TEXT( "All packages processed successfully" ) );
+    return 0;
+}
 
-        world->WorldType = EWorldType::Editor;
+bool UPerfGrapherCommandlet::RunPerfGrapher( const FString & package_name, const FMetricsParams & metrics_params ) const
```

Comment:
dont mention PerfGrapher. This tool generates some JSON, that may be used by anything

Reply:
? The whole class is named 'PerfGrapherCommandlet' ? You ask me to rename the whole class ?